xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -281,50 +451,64 @@ bool CGUIFontTTF::Load(
      m_cellHeight  _ _ _ _ _ p _ _ _ _ _ _/_ _ _ _ _  bbox.yMin, descender
 
    */
-  int cellDescender = std::min<int>(m_face->bbox.yMin, m_face->descender);
-  int cellAscender  = std::max<int>(m_face->bbox.yMax, m_face->ascender);
+
+  int cellDescender = std::min<int>(data.m_face->bbox.yMin, data.m_face->descender);
+  int cellAscender = std::max<int>(data.m_face->bbox.yMax, data.m_face->ascender);
 
   if (border)
   {
     /*
-     add on the strength of any border - the non-bordered font needs
-     aligning with the bordered font by utilising GetTextBaseLine()
-     */
-    FT_Pos strength = FT_MulFix( m_face->units_per_EM, m_face->size->metrics.y_scale) / 12;
+    add on the strength of any border - the non-bordered font needs
+    aligning with the bordered font by utilising GetTextBaseLine()
+    */
+    FT_Pos strength = FT_MulFix(data.m_face->units_per_EM, data.m_face->size->metrics.y_scale) / 12;
     if (strength < 128)
       strength = 128;
 
     cellDescender -= strength;
     cellAscender  += strength;
 
-    m_stroker = g_freeTypeLibrary.GetStroker();
-    if (m_stroker)
-      FT_Stroker_Set(m_stroker, strength, FT_STROKER_LINECAP_ROUND, FT_STROKER_LINEJOIN_ROUND, 0);
+    data.m_stroker = g_freeTypeLibrary.GetStroker();
+    if (data.m_stroker)
+      FT_Stroker_Set(data.m_stroker, strength, FT_STROKER_LINECAP_ROUND, FT_STROKER_LINEJOIN_ROUND,
+                     0);
   }
 
   // scale to pixel sizing, rounding so that maximal extent is obtained
-  float scaler  = height / m_face->units_per_EM;
+  float scaler = height / data.m_face->units_per_EM;
+
   cellDescender =
       MathUtils::round_int(cellDescender * static_cast<double>(scaler) - 0.5); // round down
   cellAscender = MathUtils::round_int(cellAscender * static_cast<double>(scaler) + 0.5); // round up
 
-  m_cellBaseLine = cellAscender;
-  m_cellHeight   = cellAscender - cellDescender;
+  data.m_cellBaseLine = cellAscender;
+  data.m_cellHeight = cellAscender - cellDescender;
+  fprintf(stderr, "data.m_cellHeight %i %i\n",data.m_cellHeight, data.m_cellBaseLine);
```

Comment:
remove?

Reply:
Still temporary, also not finished the cleanup. Thought on begin to allow only fallback to one font and not thought to become working with all :sweat_smile:.

Planned to make a separate request and to cleanup the whole Fontsystem before here comes in (without work way changes). There are much parts inside the font source files where need really a cleanup, think them from begin (maybe ~2006) and not really changed since them. Specially his singleton way with `XBMC_GLOBAL_REF(CFreeTypeLibrary, g_freeTypeLibrary);` :smirk:.